mingw redefinition warning

I get the following warning when building valhalla with mingw; for my setup I need to define the environment variable `export CXXFLAGS="-D_WIN32_WINNT=0x0600"`, but `_WIN32_WINNT` is already defined in the valhalla code.

```log
[21:29:11] /workspace/srcdir/valhalla/src/baldr/tz_alt.cpp:73:0: warning: "_WIN32_WINNT" redefined
[21:29:11]  #      define _WIN32_WINNT _WIN32_WINNT_VISTA
[21:29:11]  

```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,7 @@
    * FIXED: update vcpkg commit for Azure pipelines to fix libtool mirrors [#3915](https://github.com/valhalla/valhalla/pull/3915)
    * FIXED: fix CHANGELOG release year (2022->2023) [#3927](https://github.com/valhalla/valhalla/pull/3927)
    * FIXED: avoid segfault on invalid exclude_polygons input [#3907](https://github.com/valhalla/valhalla/pull/3907)
+   * FIXED: allow _WIN32_WINNT to be defined by build system [[#3933](https://github.com/valhalla/valhalla/issues/3933)]
 * **Enhancement**
    * CHANGED: replace boost::optional with C++17's std::optional where possible [#3890](https://github.com/valhalla/valhalla/pull/3890)
    * ADDED: parse `lit` tag on ways and add it to graph [#3893](https://github.com/valhalla/valhalla/pull/3893)

--- a/src/baldr/tz_alt.cpp
+++ b/src/baldr/tz_alt.cpp
@@ -70,10 +70,16 @@
 
 #    ifndef NTDDI_VERSION
 #      define NTDDI_VERSION 0x06000000
-#      define _WIN32_WINNT _WIN32_WINNT_VISTA
+#      ifndef _WIN32_WINNT
+#          define _WIN32_WINNT 
+#      endif
+#      ifndef _WIN32_WINNT_VISTA
+#          define _WIN32_WINNT_VISTA
+#      endif
 #    elif NTDDI_VERSION < 0x06000000
 #      warning "If this fails to compile NTDDI_VERSION may be to low. See comments above."
 #    endif
+
      // But once we define the values above we then get this linker error:
      // "tz.cpp:(.rdata$.refptr.FOLDERID_Downloads[.refptr.FOLDERID_Downloads]+0x0): "
      //     "undefined reference to `FOLDERID_Downloads'"

